Ngen build getting an BMI error

## Current behavior
Ngen build getting an error.
```#18 251.0 [ 68%] Building CXX object test/CMakeFiles/test_bmi_python.dir/realizations/catchments/Bmi_Py_Formulation_Test.cpp.o
3725
#18 255.9 [ 68%] Building CXX object test/CMakeFiles/test_bmi_multi.dir/realizations/catchments/Bmi_Cpp_Multi_Array_Test.cpp.o
3726
#18 258.0 gmake[2]: *** [CMakeFiles/ngen.dir/build.make:76: CMakeFiles/ngen.dir/src/NGen.cpp.o] Error 1
3727
#18 258.0 gmake[1]: *** [CMakeFiles/Makefile2:498: CMakeFiles/ngen.dir/all] Error 2
3728
#18 258.0 gmake[1]: *** Waiting for unfinished jobs....
```
## Expected behavior
Build without error

## Steps to replicate behavior (include URLs)

1. Disable the line 156 in Docker.ngen file.
```156  # && ./build_sub extern/test_bmi_cpp```

Recent failed action can be found here.
https://github.com/CIROH-UA/NGIAB-CloudInfra/actions/runs/6949391875

--- a/docker/Dockerfile.ngen
+++ b/docker/Dockerfile.ngen
@@ -133,6 +133,7 @@ RUN cd ${WORKDIR}/ngen \
         -DNETCDF_LIBRARY=/usr/lib/libnetcdf.so \
         -DNETCDF_CXX_INCLUDE_DIR=/usr/local/include \
         -DNETCDF_CXX_LIBRARY=/usr/local/lib64/libnetcdf-cxx4.so ; \
+        cmake --build $cmake_build_serial --target all -j $(nproc); \
     fi \
     && if [ "${BUILD_NGEN_PARALLEL}" == "true" ]; then \
         cmake -B cmake_build_parallel -S . \
@@ -149,8 +150,8 @@ RUN cd ${WORKDIR}/ngen \
         -DNETCDF_LIBRARY=/usr/lib/libnetcdf.so \
         -DNETCDF_CXX_INCLUDE_DIR=/usr/local/include \
         -DNETCDF_CXX_LIBRARY=/usr/local/lib64/libnetcdf-cxx4.so ; \
+        cmake --build $cmake_build_parallel --target all -j $(nproc);
     fi \
-    && ln -s $(if [ "${BUILD_NGEN_PARALLEL}" == "true" ]; then echo "cmake_build_parallel"; else echo "cmake_build_serial"; fi) cmake_build \
     # Build the required submodules/external libs needed for running the tests later \
     # C++ functionality isn't separate, so always build the test_bmi_cpp shared lib (also needed for test_bmi_multi) \
     # && ./build_sub extern/test_bmi_cpp \
@@ -161,8 +162,6 @@ RUN cd ${WORKDIR}/ngen \
     &&  if [ "${NGEN_ACTIVATE_FORTRAN}" == "ON" ]; then \
             ./build_sub extern/test_bmi_fortran; \
         fi \
-    &&  for BUILD_DIR in $(if [ "${BUILD_NGEN_PARALLEL}" == "true" ]; then echo "cmake_build_parallel"; fi) $(if [ "${BUILD_NGEN_SERIAL}" == "true" ]; then echo "cmake_build_serial"; fi) ; do \
-        cmake --build $BUILD_DIR --target all -j $(nproc); \
     done \
     # run the serial tests \
     && cd ${WORKDIR}/ngen \

